shared/actions/teams.js

```diff
@@ -307,65 +295,57 @@ const _inviteByEmail = function*(action: TeamsGen.InviteToTeamByEmailPayload) {
   }
 }
 
-const _addToTeam = function*(action: TeamsGen.AddToTeamPayload) {
+const addToTeam = (_, action) => {
   const {teamname, username, role, sendChatNotification} = action.payload
-  try {
-    yield* Saga.callPromise(
-      RPCTypes.teamsTeamAddMemberRpcPromise,
-      {
-        email: '',
-        name: teamname,
-        role: role ? RPCTypes.teamsTeamRole[role] : RPCTypes.teamsTeamRole.none,
-        sendChatNotification,
-        username,
-      },
-      [Constants.teamWaitingKey(teamname), Constants.addMemberWaitingKey(teamname, username)]
-    )
-  } catch (e) {
-    // identify error
-    if (e.code === RPCTypes.constantsStatusCode.scidentifysummaryerror) {
-      if (isMobile) {
-        // show profile card on mobile
-        yield Saga.put(ProfileGen.createShowUserProfile({username}))
+  return RPCTypes.teamsTeamAddMemberRpcPromise(
+    {
+      email: '',
+      name: teamname,
+      role: role ? RPCTypes.teamsTeamRole[role] : RPCTypes.teamsTeamRole.none,
+      sendChatNotification,
+      username,
+    },
+    [Constants.teamWaitingKey(teamname), Constants.addMemberWaitingKey(teamname, username)]
+  )
+    .then(() => {})
+    .catch(e => {
+      // identify error
+      if (e.code === RPCTypes.constantsStatusCode.scidentifysummaryerror) {
+        if (isMobile) {
```

Comment:
<img src="https://user-images.githubusercontent.com/11968340/50521315-162c8f00-0a93-11e9-96f6-879b1e1aa5a5.png" width="250" />
Should I be seeing this?

Reply:
unclear , nothing logically different should be happening vs master

Reply:
Hmm ok, works the same in master. I thought this was the flow where we pushed the profile